tephra-core/src/test/java/org/apache/tephra/TransactionSystemTest.java

```diff
@@ -33,25 +33,33 @@
  */
 public abstract class TransactionSystemTest {
 
-  public static final byte[] C1 = new byte[] { 'c', '1' };
-  public static final byte[] C2 = new byte[] { 'c', '2' };
-  public static final byte[] C3 = new byte[] { 'c', '3' };
-  public static final byte[] C4 = new byte[] { 'c', '4' };
+  private static final byte[] C1 = new byte[] { 'c', '1' };
+  private static final byte[] C2 = new byte[] { 'c', '2' };
+  private static final byte[] C3 = new byte[] { 'c', '3' };
+  private static final byte[] C4 = new byte[] { 'c', '4' };
 
   protected abstract TransactionSystemClient getClient() throws Exception;
 
   protected abstract TransactionStateStorage getStateStorage() throws Exception;
 
-  // Unfortunately, in-memory mode and thrift mode throw different exceptions here
-  @Test(expected = Exception.class)
+  @Test
   public void testNegativeTimeout() throws Exception {
-    getClient().startShort(-1);
+    try {
+      getClient().startShort(-1);
+      Assert.fail("Expected illegal argument for negative timeout");
+    } catch (IllegalArgumentException e) {
```

Comment:
This is because in ThriftTransactionSystemTest, we want to assert additionally that the number of retries is 0. That cannot be done if the test case has (expected=IllegalArgumentException.class).